app/src/main/java/org/wikipedia/views/ImagePreviewDialog.kt

```diff
@@ -0,0 +1,143 @@
+        super.onDestroyView()
+    }
+
+    private fun setImageDetails() {
+        loadImage(imageInfo.originalUrl)
+        titleText!!.text = StringUtil.fromHtml(fileName!!.removePrefix("File:"))
+        addDetailPortion(getString(R.string.suggested_edits_image_caption_title), StringUtil.fromHtml(imageInfo.getMetadata()!!.imageDescription()!!.value()).toString())
+        addDetailPortion(getString(R.string.suggested_edits_image_artist), StringUtil.fromHtml(imageInfo.getMetadata()!!.artist()!!.value()).toString())
+        addDetailPortion(getString(R.string.suggested_edits_image_date), imageInfo.getMetadata()!!.dateTime()!!.value())
+        addDetailPortion(getString(R.string.suggested_edits_image_source), imageInfo.getMetadata()!!.imageDescription()!!.source())
+        addDetailPortion(getString(R.string.suggested_edits_image_licensing), imageInfo.getMetadata()!!.licenseShortName()!!.value())
+        detailsHolder.requestLayout()
+    }
+
+    private fun addDetailPortion(@NonNull title: String, @Nullable detail: String?) {
+        if (!detail.isNullOrEmpty()) {
+            val view = ImageDetailView(requireContext())
+            view.setTitle(title)
+            view.setDetail(detail)
+            detailsHolder.addView(view)
+        }
+    }
+
+    override fun onDismiss(dialogInterface: DialogInterface?) {
+        super.onDismiss(dialogInterface)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == Constants.ACTIVITY_REQUEST_GALLERY && resultCode == GalleryActivity.ACTIVITY_RESULT_PAGE_SELECTED) {
+            startActivity(data)
+        } else {
+            super.onActivityResult(requestCode, resultCode, data)
+        }
+    }
+
+    private fun loadImage(url: String?) {
+        progressBar!!.visibility = View.GONE
+
+        galleryImage.visibility = View.VISIBLE
+        L.v("Loading image from url: $url")
+
+        galleryImage.controller = Fresco.newDraweeControllerBuilder()
+                .setUri(url)
+                .setAutoPlayAnimations(true)
+                .setControllerListener(object : BaseControllerListener<com.facebook.imagepipeline.image.ImageInfo>() {
+                    override fun onFinalImageSet(id: String?, imageInfo: com.facebook.imagepipeline.image.ImageInfo?, animatable: Animatable?) {
+                        galleryImage.setDrawBackground(true)
+                    }
+
+                    override fun onFailure(id: String?, throwable: Throwable?) {
+                        L.d(throwable)
+                    }
+                })
+                .build()
+    }
+
+    companion object {
+
+        fun newInstance(imageInfo: ImageInfo, fileName: String?): ImagePreviewDialog {
+            val dialog = ImagePreviewDialog()
+            val args = Bundle()
+            args.putSerializable("imageInfo", imageInfo)
+            if (fileName != null) {
+                args.putString("filename", fileName)
+            }
```

Comment:
It would be better if create `const` value for "imageInfo" and "filename"

Reply:
Done